File: 94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function inorderTraversal(root: TreeNode | null): number[] {
    let treeList: number[] = [];
    inorderHelper(root,treeList);
    return treeList;
    
};

function inorderHelper(node:TreeNode | null , treeList: number[]):number[]{
     if(!node)
        return [];
    inorderHelper(node.left,treeList); // left branch
    treeList.push(node.val)
    inorderHelper(node.right,treeList); //right branch
  
};

File: 94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public IList<int> InorderTraversal(TreeNode root) {
        IList<int> treeList = new List<int>();
        helper(root,treeList);
        return treeList;
    }
    
    public void helper(TreeNode node, IList<int> treeList){
        if(node != null){
            helper(node.left,treeList);
            treeList.Add(node.val);
            helper(node.right,treeList);
        }
        
    }
}